cdc-accelerator/feature/acls.js

```diff
@@ -0,0 +1,59 @@
+import path from 'path'
+import fs from 'fs'
+import { clearDirectoryContents } from '../utils/utils.js'
+import { SRC_DIRECTORY, BUILD_DIRECTORY } from './constants.js'
+import client from '../../cdc-accelerator/sap-cdc-toolkit/gigya/client.js'
+export default class ACL {
+    static ACL_FILE_NAME = 'Acls.json'
+    #credentials
+    constructor(credentials) {
+        this.#credentials = credentials
+    }
+
+    getName() {
+        return this.constructor.name
+    }
+
+    async init(aclID, partnerId, permissionGroupDirectory, dataCenter) {
+        const promises = aclID.map(async (ids) => {
+            const response = await this.getAclsRequest(dataCenter, ids, partnerId, this.#credentials)
+            if (response.errorCode) {
+                throw new Error(JSON.stringify(response))
+            }
+
+            return { [ids]: response['acl'] }
+        })
+        const results = await Promise.all(promises)
```

Comment:
variable promises is really a promise? it seems a regular array with defined values, you are await ing already inside the loop..